### PR TITLE
Issue/11359 unavailable cards

### PR DESCRIPTION
**Pull request by hichamboushaba**
Part of: #11359 

⚠️ Don't merge until https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2992 is merged and FluxC version updated.

### Description
This PR includes the following changes:
1. Update the DashboardWidget model to include an availability flag (we might need to improve this to support different statuses, and not just availability, but this can be done in a separate PR)
2. Re-work the `DragAndDropItemsList`:
    - Make the component more generic and accepting custom UIs for the items.
    - Allow marking some items as non-draggable.
3. Update the Widget settings screen to handle unavailable widgets.
4. Mark the `Performance` and `Top Performers` widgets as unavailable when the site doesn't have orders.

Reviewing bc5845cb39b1f8388cef4da66d8a8edf02b26d88 (about the `DragAndDropItemsList`) separately would make the review of other parts easier.

I'll handle the other cards (Blaze and the Onboarding) in separate PRs.

### Testing instructions
1. Create a new Jurassic Ninja site.
2. Open the app and sign in to the site.
3. Open the dashboard screen and confirm the Performance and Top Performers cards are hidden.
4. Open the widget editor screen and confirm both cards are marked as unavailable.
5. Test drag and dropping, and confirm it works as expected.
6. Create some orders.
7. Confirm the cards are shown.

### Images/gif

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/analytics/hub/settings/AnalyticsHubSettingScreen.kt [hunk 1]
```diff
@@ -28,59 +33,137 @@ import androidx.compose.ui.unit.dp
...
+                    Text(
+                        text = stringResource(id = R.string.save).uppercase()
+                    )
+                }
+            },
+        )
+    }) { padding ->
+        when (state) {
+            is AnalyticsHubSettingsViewState.CardsConfiguration -> {
+                DragAndDropItemsList(
```
Comment (hichamboushaba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/analytics/hub/settings/AnalyticsHubSettingScreen.kt:95]: @atorresveiga if you have some time, please review the commit bc5845cb39b1f8388cef4da66d8a8edf02b26d88, it introduces a more generic component that can be used by both the Analytics Hub screen and the Dashboard.

This also fixes an issue where other rows were draggable below the `ExplorePlugin`, now it's fixed at the bottom of the list.

**Review by 0nko — COMMENTED** [review]
Heavy PR to wrap my head around.. 😅 Really nice job making everything work. I tested the functionality and it worked well but I noticed that there is some blank space in the dashboard when widgets are unavailable:

image

**Comment by hichamboushaba** [comment]
Thanks @0nko for the review, and sorry for the size of the PR.
Regarding the issue you mentioned, assuming by the screenshot I think the cause is that we are not showing any widgets, as in the branch both the onboarding and the top performers cards are not shown in the container yet, but they are both shown on their own ComposeViews, so the white space you are seeing is just the container's padding, and shouldn't be an issue, since we'll necessary have a single widget available at minimum.

**Review by 0nko — APPROVED** [review]
Nice work! 🥇

**Pull request merged**